CLI/src/Serialization.cs

```diff
@@ -6,111 +6,230 @@
+    /// <summary>
+    /// ConfigSerializer class to encode/decode config object between client & server communication. 
+    /// </summary>
+    public class ConfigSerializer : Serializer<Config>
+    {
+        internal override void Encode(Config data) { Encode(msgpack, data); }
+        internal override Config Decode() { return Decode(msgpack); }
+
+        internal static void Encode(MsgPack msgpack, Config data)
+        {
+            msgpack.ForcePathObject("ErrorFile").AsString = string.IsNullOrEmpty(data.ErrorFile)?string.Empty:data.ErrorFile;
+            msgpack.ForcePathObject("skeletonPath").AsString = string.IsNullOrEmpty(data.skeletonPath) ? string.Empty : data.skeletonPath;
+            msgpack.ForcePathObject("Codegen").AsString = Convert.ToString(data.Codegen);
+            msgpack.ForcePathObject("EncFormat").AsString = string.IsNullOrEmpty(data.EncFormat) ? string.Empty : data.EncFormat;
+
+            MsgPack item;
+            item = msgpack.ForcePathObject("InputFiles");
+
+            if (data.InputFiles != null && data.InputFiles.Count > 0)
+            {
+                foreach (string inputFile in data.InputFiles)
+                {
+                    var child = item.AddArrayChild();
+                    child.ForcePathObject("InputFile").AsString = inputFile;
+                }
+            }
+
+            item = msgpack.ForcePathObject("OutputFiles");
+
+            if (data.OutputFiles !=null && data.OutputFiles.Count>0)
+            {
+                foreach (string outputFile in data.OutputFiles)
+                {
+                    var child = item.AddArrayChild();
+                    child.ForcePathObject("OutputFile").AsString = outputFile;
+                }
+            }
+
+            item = msgpack.ForcePathObject("CopyFolders");
+
+            if (data.CopyFolders != null && data.CopyFolders.Count > 0)
+            {
+                foreach (string copyFolder in data.CopyFolders)
+                {
+                    var child = item.AddArrayChild();
+                    child.ForcePathObject("CopyFolder").AsString = copyFolder;
+                }
+            }
+
+            item = msgpack.ForcePathObject("Copies");
+
+            if (data.Copies != null && data.Copies.Count > 0)
+            {
+                foreach (string copy in data.Copies)
+                {
+                    var child = item.AddArrayChild();
+                    child.ForcePathObject("Copy").AsString = copy;
+                }
+            }
+        }
+        internal static Config Decode(MsgPack msgpack)
+        {
+            Config config = new Config();
+            config.ErrorFile = (string)msgpack.ForcePathObject("ErrorFile").AsString;
+            config.skeletonPath = (string)msgpack.ForcePathObject("skeletonPath").AsString;
+            config.Codegen = ((string)msgpack.ForcePathObject("Codegen").AsString).Equals("True") ? true : false;
```

Comment:
Config.Codegen accepts bool value.  Hence, if "True" is returned as string, converting it to true as boolean, else false.